dataprofiler/profilers/utils.py

```diff
@@ -483,3 +483,33 @@ def find_diff_of_dicts(dict1, dict2):
         diff = "unchanged"
 
     return diff
+
+def get_pairwise_corr(data1, data2, mean1, mean2, std1, std2):
+    """
+    Computes the pairwise correlation of the data if the mean and
+    unbiased standard deviation of the respective data is known.
+    One-pass calculation.
+
+    :param data1: The first set of data
+    :type data1: pd.Dataframe
+    :param data2: The second set of data
+    :type data2: pd.Dataframe
+    :param mean1: The mean of the first data
+    :type mean1: float
+    :param mean2: The mean of the second set of data
+    :type mean2: float
+    :param std1: The stddev of the first set of data
+    :type std1: float
+    :param std2: The stddev of the second set of data
+    :type std2: float
+    :return: The correlation
+    :rtype: float
+    """
+    covariance = 0
+    for i1, i2 in zip(data1, data2):
+        if (not np.isnan(i1)) and (not np.isnan(i2)):
+            a = i1 - mean1
+            b = i2 - mean2
+            covariance += a * b
+
+    return (covariance / (len(data1.index) - 1)) / (std1 * std2)
```

Comment:
is this not still ignoring the data points which have NaNs.

In otherwords if a value in a row is NaN for col1, but a real value in col2, shouldn't it be `mean1 * i2`?

Also, wouldn't std1 change as a result?

Reply:
Additionally, I think instead of looping ourselves, we might be able to utilize vectorization to do this more quickly.
e..g
1. replace all Nans (in a copy of the dataset)
2. compute: `sum(fixed_column1 * fixed_column2)`
3. compute: updated stddev for columns which had updated nans (not actual std of new column, just updating relative to nan values (if possible) maybe this is expensive?

Thoughts?

Reply:
The function works as expected with handling means. The covariance can be defined as `sum((x_i - mean_x)(y_i - mean_y))/n`. So, if there's a NaN value and it's imputed to the mean of one of the columns, that term in the summation is zero and can be ignored when looping through no matter what the value in the other column is.

I agree that there is some confusion on std that could be addressed. I calculate the updated std prior to calling the function, so the function itself already receives an updated std. It would make more sense to update the std within this function for added clarity and improve the flexibility of this function, so I will make those changes.

Vectorization is also an idea that can definitely be implemented to improve the speed of the function. I'll play around with this.

Reply:
If we know the location of the NaNs from the previous calculations, could we technically drop them without looping and then do the multiplication?

Reply:
That way we don't have to call `np.isnan` again, this value may not be representative of all which the user want's considered a `NaN`